### PR TITLE
linstor: Fix volume download and make resource available on copy target

### PR DESCRIPTION
**Pull request by rp-**
### Description

Linstor primary storage forgot to make sure
the volume download/copy target has a Linstor resource available.

### Types of changes

- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] Enhancement (improves an existing feature and functionality)
- [ ] Cleanup (Code refactoring and cleanup, that may add test cases)
- [ ] build/CI

#### Feature/Enhancement Scale

- [ ] Major
- [ ] Minor

#### Bug Severity

- [ ] BLOCKER
- [ ] Critical
- [x] Major
- [ ] Minor
- [ ] Trivial

### How Has This Been Tested?

Tested on hyperconverged and non-hyperconverged 4.19.0 Cloudstack.

### CODE REVIEW

File: plugins/storage/volume/linstor/src/main/java/org/apache/cloudstack/storage/datastore/driver/LinstorPrimaryDataStoreDriverImpl.java [hunk 1]
```diff
@@ -965,6 +981,36 @@ private Answer copyTemplate(DataObject srcData, DataObject dstData) {
...
     }
 
+    private Answer copyVolume(DataObject srcData, DataObject dstData) {
+        VolumeInfo srcVolInfo = (VolumeInfo) srcData;
+        final StoragePoolVO pool = _storagePoolDao.findById(srcVolInfo.getDataStore().getId());
+        final DevelopersApi api = LinstorUtil.getLinstorAPI(pool.getHostAddress());
+        final String rscName = LinstorUtil.RSC_PREFIX + srcVolInfo.getUuid();
+
+        int nMaxExecutionMinutes = NumbersUtil.parseInt(
+                _configDao.getValue(Config.SecStorageCmdExecutionTimeMax.key()), 30);
```
Comment (sureshanaparti) [inline plugins/storage/volume/linstor/src/main/java/org/apache/cloudstack/storage/datastore/driver/LinstorPrimaryDataStoreDriverImpl.java:991]: does this copy performed by ssvm? check if this is suitable config - `copy.volume.wait`

Comment (rp-) [reply plugins/storage/volume/linstor/src/main/java/org/apache/cloudstack/storage/datastore/driver/LinstorPrimaryDataStoreDriverImpl.java:991]: TBH, I don't know exactly where the actual copy ends up.
I think it is used in the KVMStorageProcessor and will just use qemu-img convert to move it into secondary storage from Linstor.
So is here the `SecStorageCmdExecutionTimeMax` or `copy.volume.wait` more fitting?

Comment (DaanHoogland) [reply plugins/storage/volume/linstor/src/main/java/org/apache/cloudstack/storage/datastore/driver/LinstorPrimaryDataStoreDriverImpl.java:991]: default for `SecStorageCmdExecutionTimeMax` is 30 minutes.
default for `CopyVolumeWait` is 10800 secs (180 minutes)

by the name of your method `CopyVolumeWait` seems more appropriate, but you could even create your own var (or use either)

**Review by DaanHoogland — APPROVED** [review]
clgtm

File: plugins/storage/volume/linstor/src/main/java/org/apache/cloudstack/storage/datastore/driver/LinstorPrimaryDataStoreDriverImpl.java (hunk 1 above)
Comment (sureshanaparti) [reply plugins/storage/volume/linstor/src/main/java/org/apache/cloudstack/storage/datastore/driver/LinstorPrimaryDataStoreDriverImpl.java:991]: > `copy.volume.wait`

`copy.volume.wait` seems appropriate

Comment (rp-) [reply plugins/storage/volume/linstor/src/main/java/org/apache/cloudstack/storage/datastore/driver/LinstorPrimaryDataStoreDriverImpl.java]: done

**Comment by DaanHoogland** [comment]
@bluorangutan package

**Comment by rp-** [comment]
@DaanHoogland 
I have to rework this, with my findings about format and template from snapshots.

**Comment by DaanHoogland** [comment]
> @bluorangutan package
but now without typo:
@blueorangutan package

**Comment by DaanHoogland** [comment]
@blueorangutan test alma9 kvm-alma9

**Review by DaanHoogland — APPROVED** [review]
still clgtm, let's wait for smoke tests

**Comment by DaanHoogland** [comment]
@rp- is it done now?

**Comment by rp-** [comment]
> @rp- is it done now?

yes

**Comment by rp-** [comment]
@DaanHoogland
can we merge this?

**Comment by DaanHoogland** [comment]
no ;)

**Pull request merged**